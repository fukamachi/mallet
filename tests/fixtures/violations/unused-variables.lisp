;;; Test file for unused-variables rule

;; Unused function parameters
(defun add (x y)
  (+ x 1))

;; Unused let binding
(let ((a 1) (b 2))
  (+ a 10))

;; Unused loop variable
(loop for item in list
      for index from 0
      collect item)

;; Valid: all used
(defun multiply (x y)
  (* x y))

;; Valid: explicitly ignored
(defun process (data _options)
  (print data))

;; Valid: declare ignore
(defun compute (x y)
  (declare (ignore y))
  (+ x 1))
