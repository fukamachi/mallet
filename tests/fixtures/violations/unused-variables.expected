# Expected violations for unused-variables.lisp
# Format: line:column rule-name severity

4:14 unused-variables warning
8:13 unused-variables warning
13:10 unused-loop-variables info
